wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -2405,17 +2415,36 @@ instance BuildableSafeGen SlotDuration where
 data NodeSettings = NodeSettings {
      setSlotDuration   :: !SlotDuration
    , setSoftwareInfo   :: !(V1 Core.SoftwareVersion)
-   , setProjectVersion :: !Version
+   , setProjectVersion :: !(V1 Version)
    , setGitRevision    :: !Text
    } deriving (Show, Eq, Generic)
 
 #if !(MIN_VERSION_swagger2(2,2,2))
 -- See note [Version Orphan]
-instance ToSchema Version where
+instance ToSchema (V1 Version) where
     declareNamedSchema _ =
-        pure $ NamedSchema (Just "Version") $ mempty
+        pure $ NamedSchema (Just "V1Version") $ mempty
             & type_ .~ SwaggerString
 
+instance Buildable (V1 Version) where
+    build (V1 v) = bprint shown v
+
+instance Buildable (SecureLog (V1 Version)) where
+    build (SecureLog x) = Formatting.Buildable.build x
+
+instance ToJSON (V1 Version) where
+    toJSON (V1 v) = toJSON (showVersion v)
+
+instance FromJSON (V1 Version) where
+    parseJSON (String v) = case readP_to_S parseVersion (T.unpack v) of
+        (reverse -> ((ver,_):_)) -> pure (V1 ver)
+        _                        -> mempty
+    parseJSON x                  = typeMismatch "Not a valid Version" x
```

Comment:
What's with the `reverse` :confused: ?

```hs
   parseJSON (String v) = case readP_to_S parseVersion (T.unpack v) of
        [(ver, _)] -> pure (V1 ver)
        _          -> mempty
```

Isn't that enough ?

Reply:
```
Prelude> import Data.Version
Prelude Data.Version> let v = makeVersion [1,2,3]
Prelude Data.Version> v
Version {versionBranch = [1,2,3], versionTags = []}
Prelude Data.Version> import Text.ParserCombinators.ReadP
Prelude Data.Version Text.ParserCombinators.ReadP> let out = readP_to_S parseVersion (showVersion v)
Prelude Data.Version Text.ParserCombinators.ReadP> out
[(Version {versionBranch = [1], versionTags = []},".2.3"),(Version {versionBranch = [1,2], versionTags = []},".3"),(Version {versionBranch = [1,2,3], versionTags = []},"")]
```
I need the last element. And I can have [1,2] also

Reply:
Arf. Okay.